File: SWB4/Client/OpenOffice/OfficeComunicationInterfaces/src/org/semanticwb/office/interfaces/ElementInfo.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package org.semanticwb.office.interfaces;

/**
 *
 * @author victor.lorenzana
 */
public class ElementInfo {
    public String id;    
    public String title;
    public boolean active;
    public String type;

    @Override
    public boolean equals(Object obj)
    {
        if (obj == null)
        {
            return false;
        }
        if (getClass() != obj.getClass())
        {
            return false;
        }
        final ElementInfo other = (ElementInfo) obj;
        if ((this.id == null) ? (other.id != null) : !this.id.equals(other.id))
        {
            return false;
        }
        if ((this.type == null) ? (other.type != null) : !this.type.equals(other.type))
        {
            return false;
        }
        return true;
    }

    @Override
    public int hashCode()
    {
        int hash = 3;
        hash = 97 * hash + (this.id != null ? this.id.hashCode() : 0);
        hash = 97 * hash + (this.type != null ? this.type.hashCode() : 0);
        return hash;
    }
    

    @Override
    public String toString()
    {
        return title.toString();
    }

}

File: SWB4/Client/Microsoft Office/INFOTEC WebBuilder 4/WBOffice4/Interfaces/ElementInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace WBOffice4.Interfaces
{
    public class ElementInfo
    {
        public String id;
        public String title;
        public bool active;
        public String type;


        public override bool Equals(Object obj)
        {
            if (obj == null)
            {
                return false;
            }
            if (GetType() != obj.GetType())
            {
                return false;
            }
            ElementInfo other = (ElementInfo)obj;
            if ((this.id == null) ? (other.id != null) : !this.id.Equals(other.id))
            {
                return false;
            }
            if ((this.type == null) ? (other.type != null) : !this.type.Equals(other.type))
            {
                return false;
            }
            return true;
        }


        public override int GetHashCode()
        {
            int hash = 3;
            hash = 97 * hash + (this.id != null ? this.id.GetHashCode() : 0);
            hash = 97 * hash + (this.type != null ? this.type.GetHashCode() : 0);
            return hash;
        }



        public override String ToString()
        {
            return title.ToString();
        }
    }
}
